lib/bundler/fetcher/compact_gem_list.rb

```diff
@@ -0,0 +1,59 @@
+module Bundler
+  class Fetcher
+    class CompactGemList
+      require 'set'
+
+      autoload :Cache,   'bundler/fetcher/compact_gem_list/cache'
+      autoload :Updater, 'bundler/fetcher/compact_gem_list/updater'
+
+      attr_reader :fetcher, :directory
+
+      def initialize(fetcher, directory)
+        @fetcher = fetcher
+        @directory = Pathname(directory)
+        FileUtils.mkdir_p(@directory)
+        @updater = Updater.new(@fetcher)
+        @cache   = Cache.new(@directory)
+        @endpoints = Set.new
+      end
+
+      def names
+        update([[@cache.names_path, 'names']])
+        @cache.names
+      end
+
+      def versions
+        update([[@cache.versions_path, 'versions']])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.